internal/catalog/internal/types/health_status.go

```diff
@@ -27,6 +28,63 @@ func RegisterHealthStatus(r resource.Registry) {
 	r.Register(resource.Registration{
 		Type:     HealthStatusV1Alpha1Type,
 		Proto:    &pbcatalog.HealthStatus{},
-		Validate: nil,
+		Validate: ValidateHealthStatus,
 	})
 }
+
+func ValidateHealthStatus(res *pbresource.Resource) error {
+	var hs pbcatalog.HealthStatus
+
+	if err := res.Data.UnmarshalTo(&hs); err != nil {
+		return newErrDataParse(&hs, err)
+	}
+
+	var err error
+
+	// Should we allow empty types? I think for now it will be safest to require
+	// the type field is set and we can relax this restriction in the future
+	// if we deem it desirable.
+	if hs.Type == "" {
+		err = multierror.Append(err, ErrInvalidField{
+			Name:    "type",
+			Wrapped: errMissing,
+		})
+	}
+
+	// Ensure that the HealthStatus' owner is a type that we want to allow. The
+	// owner is currently the resource that this HealthStatus applies to. If we
+	// change this to be a parent reference within the HealthStatus.Data then
+	// we could allow for other owners.
+	if res.Owner == nil {
+		err = multierror.Append(err, ErrInvalidField{
+			Name:    "owner",
+			Wrapped: errMissing,
+		})
+	} else if !canAssociateHealthStatus(res.Owner) {
+		err = multierror.Append(err, ErrOwnerInvalid{ResourceType: res.Id.Type, OwnerType: res.Owner.Type})
+	}
+
+	return err
+}
+
+func canAssociateHealthStatus(id *pbresource.ID) bool {
+	// Ensure that only other resources within this group
+	// can own the resource.
+	if id.Type.Group != GroupName {
+		return false
+	}
+
+	// As these checks are performed at admission time we can assume
+	// resource version translation will already have occurred and
+	// we can enforce that the group version is the current version.
+	if id.Type.GroupVersion != CurrentVersion {
+		return false
+	}
+
+	// ensure the only owners can be Workloads or Nodes
+	if id.Type.Kind == WorkloadKind || id.Type.Kind == NodeKind {
+		return true
+	}
+
+	return false
+}
```

Comment:
It feels like we might want an `EqualType` method in the `internal/resource` package that doesn't use reflection. Then I guess this could be:

```Go
return resource.EqualType(id.Type, WorkloadV1Alpha1Type) ||
       resource.EqualType(id.Type, NodeV1Alpha1Type)
```

Reply:
I dropped an EqualType method into the internal/resource/utils package I created and am now using that.